Update getting started: Session Replay (#6390)
PostHog/posthog.com: contents/docs/session-replay/manual.mdx

Above is a recording of posthog.com… on posthog.com – very meta.
</Caption>

## Step one: Install our JavaScript web library

import Snippet from "../../../integrate/\_snippets/install-js-snippet.mdx"
import Badge from "../../../getting-started/\_snippets/badge.mdx"
import InstallJSViaNPM from "../../../integrate/\_snippets/install-js-via-npm.mdx"

### Option one: Add JavaScript snippet to your HTML <Badge>Recommended</Badge>

<Snippet />

### Option two: Install via NPM

<InstallJSViaNPM />

## Step two: Enable session recordings in your project settings

Enable session recordings in your PostHog [Project Settings](https://app.posthog.com/project/settings).

![Enable session recordings in your PostHog](../../images/docs/session-replay/enable-session-replay-in-project-settings.png)

Once enabled, the library will start recording sessions by default.

It can be toggled on and off in the JS library by appropriately setting the `disable_session_recording` flag in the [config](/docs/integrate/client/js/#config). 

Users who [opt out](/docs/libraries/js#opt-users-out) of event capturing will not have their sessions recorded.

> > **Note on using Segment's SDK**: Session Replay does not work if you send data using Segment's SDK as this data is not collected. If you use Segment, you may want to add the PostHog library too (make sure to only send regular event data from one source).

Reviewer: Could we link to the docs that detail how to get it working with Segment?

Above is a recording of posthog.com… on posthog.com – very meta.
</Caption>

## Step 1: Install our JavaScript web library

First, you need to install our JavaScript library. You can do so by either:

import Snippet from "../../../integrate/\_snippets/install-js-snippet.mdx"
import Badge from "../../../getting-started/\_snippets/badge.mdx"

> import InstallJSViaNPM from "../../../integrate/\_snippets/install-js-via-npm.mdx"

Reviewer 2: Should we import these at the very top of the file?

Author: Perhaps. Its not necessary though. More of a styling issue

### Option one: Add JavaScript snippet to your HTML <Badge>Recommended</Badge>

<Snippet />

> ### Option two: Install via NPM

Reviewer 2: This is not just NPM, "install via package manager"?

<InstallJSViaNPM />

## Step 2: Enable recordings in your project settings

> Enable session recordings in your PostHog [Project Settings](https://app.posthog.com/project/settings).

Reviewer 2: ```suggestion
Enable session recordings in your PostHog [project settings](https://app.posthog.com/project/settings).
```

![Enable session recordings in your PostHog](../../images/docs/session-replay/enable-session-replay-in-project-settings.png)

> Once enabled, the library will start recording sessions by default.

Reviewer 2: Should we mention something about sessions recordings being blocked and reverse proxy?

Author: We could. There's a "troubleshooting and FAQ section", so I think that's sufficient for now

> It can be toggled on and off in the JS library by appropriately setting the `disable_session_recording` flag in the [config](/docs/integrate/client/js/#config). 

Reviewer 2: ```suggestion
They can be toggled off in the by setting the `disable_session_recording: true` flag in the [config](/docs/integrate/client/js/#config). 
```
Technically you can also set that on the snippet and you wouldn't enabled session replays by setting `disable_session_recording: false`
